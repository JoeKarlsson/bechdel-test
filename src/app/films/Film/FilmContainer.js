import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Film from './Film';
import api from '../../helper/api';

class FilmContainer extends Component {
	constructor() {
		super();
		this.state = {
			film: {
				title: '',
				images: {
					poster: '',
					backdrop: '',
				},
				plot: '',
				directors: [],
				writers: [],
				genres: [],
				rated: '',
				bechdelResults: {
					pass: false,
					bechdelScore: 0,
					numScenesPass: 0,
					scenesThatPass: [],
					numScenesDontPass: 0,
					numOfFemalesChars: 0,
					numOfMaleChars: 0,
					numOfFemalesCharsWithDialogue: 0,
					numOfMaleCharsWithDialogue: 0,
					totalLinesFemaleDialogue: 0,
					totalLinesMaleDialogue: 0,
				},
			},
			loading: true,
		};

		this.getFilm = this.getFilm.bind(this);
	}

	componentDidMount() {
		try {
			this.getFilm();
		} catch (error) {
			console.log(error);
		}
	}

	getFilm() {
		const { id } = this.props.match.params;
		const url = `/api/film/${id}`;
		const options = {
			method: 'GET',
		};

		this.setState({
			loading: true,
		});

		api(url, options)
			.then(data => {
				// console.log(data);
				this.setState({
					film: data,
					loading: false,
				});
				return data;
			})
			.catch(err => {
				console.error(err);
				this.setState({
					loading: false,
				});
			});
	}

	render() {
		return <Film {...this.state} />;
	}
}

FilmContainer.propTypes = {
	match: PropTypes.shape({
		params: PropTypes.shape({
			id: PropTypes.string,
		}),
	}).isRequired,
};

FilmContainer.defaultProps = {};

export default FilmContainer;
