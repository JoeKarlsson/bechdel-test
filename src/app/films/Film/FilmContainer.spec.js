import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import FilmContainer from './FilmContainer';

jest.mock('../../helper/api');

configure({ adapter: new Adapter() });

describe('FilmContainer', () => {
	let wrapper;
	let inst;
	const router = {
		params: {
			id: 1234,
		},
	};

	beforeEach(() => {
		wrapper = shallow(<FilmContainer match={router} />);
		inst = wrapper.instance();
	});

	describe('rendering', () => {
		describe('initial state', () => {
			it('should render zero state correctly', () => {
				const component = renderer.create(<FilmContainer match={router} />);
				const tree = component.toJSON();
				expect(tree).toMatchSnapshot();
			});
			it('is rendered correctly', () => {
				expect(wrapper).toHaveLength(1);
			});
			it('should not have any inital props', () => {
				const initialProps = inst.props;
				const expectedProps = {
					match: {
						params: {
							id: 1234,
						},
					},
				};
				expect(initialProps).toMatchObject(expectedProps);
			});
			it('should not have any inital state', () => {
				const initialState = inst.state;
				const expectedState = {
					film: {
						title: '',
						images: {
							poster: '',
							backdrop: '',
						},
						plot: '',
						directors: [],
						writers: [],
						genres: [],
						rated: '',
						bechdelResults: {
							pass: false,
							bechdelScore: 0,
							numScenesPass: 0,
							scenesThatPass: [],
							numScenesDontPass: 0,
							numOfFemalesChars: 0,
							numOfMaleChars: 0,
							numOfFemalesCharsWithDialogue: 0,
							numOfMaleCharsWithDialogue: 0,
							totalLinesFemaleDialogue: 0,
							totalLinesMaleDialogue: 0,
						},
					},
				};
				expect(initialState).toMatchObject(expectedState);
			});
		});
	});
});
