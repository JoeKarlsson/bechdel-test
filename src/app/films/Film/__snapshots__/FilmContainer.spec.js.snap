// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmContainer rendering initial state should render zero state correctly 1`] = `
<div>
  <span
    className="Loader"
  >
    <div
      className="Loader_indicator"
    >
      <h1>
        <span>
          Destroying the Patriarchy
        </span>
        <span
          className="Loader_ellipsis"
        >
          <span
            className="Loader_ellipsisDot"
          >
            .
          </span>
          <span
            className="Loader_ellipsisDot"
          >
            .
          </span>
          <span
            className="Loader_ellipsisDot"
          >
            .
          </span>
        </span>
      </h1>
    </div>
  </span>
</div>
`;
